src/main/java/net/starlark/java/eval/Sequence.java

```diff
@@ -41,6 +41,28 @@ default boolean truth() {
     return !isEmpty();
   }
 
+  /**
+   * The number of elements in the collection.
+   *
+   * @throws IllegalStateException if collection size exceeds int32 range
```

Comment:
I do not wish to complicate our core interfaces to support such an obscure case. It's impossible to create a StarlarkList of more than maxint32 elements because of unfortunate limitations of the JVM architecture. Thus, any attempt to create a list larger than that must fail dynamically. That's ok: no Starlark user needs a list that big. 

We can take the same approach with range lists, and cause construction to fail if any parameters would exceed maxint32.